subito-searcher.py

```diff
@@ -63,37 +64,74 @@ def run_query(url, name):
     page = requests.get(url)
     soup = BeautifulSoup(page.text, 'html.parser')
 
-    products_list = soup.find(class_='items_listing')
-    product_list_items = products_list.find_all(class_='item_list_inner')
+    # find the father of the visible items
+    text = re.findall("jsx-[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9] items visible", page.text)
+    products_list = soup.find(class_=text)
+
     msg = []
 
-    for product in product_list_items:
-        desc = product.find(class_='item_description')
-        title = desc.find('a').contents[0]
-        if(desc.find(class_='item_price') is not None):
-            price = desc.find(class_='item_price').contents[0]
-        else:
-            price = "Unknown price"
-        link = desc.find('a').get('href')
-        location = desc.find(class_="item_location").contents[0]
-
-        if not queries.get(name):   # insert the new search
-            queries[name] = {url: {link: {'title': title, 'price': price, 'location': location}}}
-            print("\nNew search added:", name)
-            print("Adding result:", title, "-", price, "-", location)
-        else:   # add search results to dictionary
-            if not queries.get(name).get(url).get(link):   # found a new element
-                tmp = "New element found for "+name+": "+title+" @ "+price+" - "+location+" --> "+link
-                msg.append(tmp)
-                queries[name][url][link] = {'title': title, 'price': price, 'location': location}
+    for products_list_index in products_list:
+        link = products_list_index.get('href')
+
+        text = "AdElements__Item--wrapper-\w*\""
+        text = text.replace("\"", "")
+        text = re.findall(text, str(products_list_index.contents))
```

Comment:
Sorry, I don't get it. Why would you define a pattern string, replace the escapes and findall when you could just match "AdElements__Item--wrapper-*" ? 
Have you tried if this works? it would be much cleaner to read

Reply:
As I mentionated in the above comment, the regex didn't work and when finally it returns the correct result, I left the replace instruction and repeated then in every part of the code.